src/compiler/scala/tools/nsc/transform/async/TransformUtils.scala

```diff
@@ -89,7 +89,7 @@ private[async] trait TransformUtils extends AsyncTransformStates {
 
   def literalUnit: Tree = Literal(Constant(())).setType(definitions.UnitTpe) // a def to avoid sharing trees
   def literalBoxedUnit: Tree = gen.mkAttributedRef(definitions.BoxedUnit_UNIT)
-  def literalBool(b: Boolean): Tree = Literal(Constant(b)).setType(definitions.BooleanTpe)
+  def literalBool[B <: Boolean](b: B): Tree = Literal(Constant(b)).setType(definitions.BooleanTpe)
```

Comment:
Is there a reason not to use `@deprecatedName` here?

Reply:
this was a case where this works just as well (since it is boxed for Constant, assuming it is not reboxed), so this was the experiment.